testsupport/wait/host.go

```diff
@@ -704,6 +713,99 @@ func containsUserAccountStatus(uaStatuses []toolchainv1alpha1.UserAccountStatusE
 	return false
 }
 
+// WaitForUserTier waits until an UserTier with the given name exists and matches any given criteria
+func (a *HostAwaitility) WaitForUserTier(name string, criteria ...UserTierWaitCriterion) (*toolchainv1alpha1.UserTier, error) {
+	a.T.Logf("waiting until UserTier '%s' in namespace '%s' matches criteria", name, a.Namespace)
+	tier := &toolchainv1alpha1.UserTier{}
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		obj := &toolchainv1alpha1.UserTier{}
+		err = a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj)
+		if err != nil && !errors.IsNotFound(err) {
+			// return the error
+			return false, err
+		} else if errors.IsNotFound(err) {
+			// keep waiting
+			return false, nil
+		}
+		tier = obj
+		return matchUserTierWaitCriterion(obj, criteria...), nil
+	})
+	// no match found, print the diffs
+	if err != nil {
+		a.printUserTierWaitCriterionDiffs(tier, criteria...)
+	}
+	return tier, err
+}
+
+// UserTierWaitCriterion a struct to compare with an expected UserTier
+type UserTierWaitCriterion struct {
+	Match func(*toolchainv1alpha1.UserTier) bool
+	Diff  func(*toolchainv1alpha1.UserTier) string
+}
+
+func matchUserTierWaitCriterion(actual *toolchainv1alpha1.UserTier, criteria ...UserTierWaitCriterion) bool {
+	for _, c := range criteria {
+		// if at least one criteria does not match, keep waiting
+		if !c.Match(actual) {
+			return false
+		}
+	}
+	return true
+}
+
+func (a *HostAwaitility) printUserTierWaitCriterionDiffs(actual *toolchainv1alpha1.UserTier, criteria ...UserTierWaitCriterion) {
+	buf := &strings.Builder{}
+	if actual == nil {
+		buf.WriteString("failed to find UserTier\n")
+	} else {
+		buf.WriteString("failed to find UserTier with matching criteria:\n")
+		buf.WriteString("actual:\n")
+		y, _ := StringifyObject(actual)
+		buf.Write(y)
+		buf.WriteString("\n----\n")
+		buf.WriteString("diffs:\n")
+		for _, c := range criteria {
+			if !c.Match(actual) {
+				buf.WriteString(c.Diff(actual))
+				buf.WriteString("\n")
+			}
+		}
+	}
+	// also include other resources relevant in the host namespace, to help troubleshooting
+	buf.WriteString(a.sprintAllResources())
```

Comment:
I don't think that we need to print all resources when the UserTier is not found. We could maybe all UserTiers which can be done in the `WaitForUserTier` function:
```
if err != nil {
	a.printUserTierWaitCriterionDiffs(tier, criteria...)
	a.listAndPrint("UserTier", a.Namespace, &toolchainv1alpha1.UserTierList{})
}
```
```suggestion
```

TBH. I think that we can drop this `sprintAllResources` completely and replace it with the `listAndPrint` approach, but that can be done as a separate effort.